content/sdk/10.js/00.ethers/20.v6/11.migration.md

```diff
@@ -1,32 +1,140 @@
+- This migration guide applies to `zksync-ethers v6`.
+- If you're using v5, no changes are required in your code as `zksync-ethers v5` uses `ethers v5`, similar to the
+deprecated `zksync-web3`.
 ::
 
-This guide provides some high-level differences between `zksync-web3`/`zksync-ethers v5` and `zksync-ethers v6`
-for those who are familiar with `zksync-web3` and need a quick primer. `zksync-ethers v6` has several differences
-since it's using `ethers.js v6`, compared to `zksync-web3`, which uses `ethers.js v5`. Before continuing to get
-to know the changes made in `zksync-ethers`, you should read the
-[migration guide](https://docs.ethers.org/v6/migrating/) from `ethers.js v5` to `ethers.js v6`.
-
-Differences:
-
-- `Token.Address` has been removed because it was deprecated.
-- `Provider.getMessageProof` has been removed because it was deprecated.
-- `Provider.getTokenPrice` has been removed because it was deprecated.
-- `Provider.getBlockWithTransaction` has been replaced with `Provider.getBlock(<BlockTag>, true)`.
-- `BlockWithTransaction` has been removed.
-- `TransactionRequest.calldata` has been changed from `BytesLike` to `string`.
-- `transaction.calldata` parameter in`Provider.estimateL1ToL2Execute` has been changed from `BytesLike` to `string`.
-- `transaction.calldata` parameter in `AdaterL1.getRequestExecuteTx` has been changed from `BytesLike` to `string`.
-- `transaction.calldata` parameter in `AdapterL1.estimateGasRequestExecute` has been changed from `BytesLike` to `string`.
-- `transaction.calldata` parameter in `AdapterL1.RequestExecute` has been changed from `BytesLike` to `string`.
-- `utils.parseTransaction` has been replaced by `utils.parseEip712`.
+This guide highlights the key differences between `zksync-web3`/`zksync-ethers v5` and `zksync-ethers v6` for those
+familiar with `zksync-web3` and need a quick primer. `zksync-ethers v6` incorporates several changes due to its
+integration with `ethers.js v6`. Before proceeding, it's recommended to read
+the [migration guide](https://docs.ethers.org/v6/migrating/) from `ethers.js v5` to `ethers.js v6`.
+
+### Key differences
+
+- **Removal of deprecated functions and properties:**
+  - `Token.Address` has been removed.
+  - `Provider.getMessageProof` has been removed.
+  - `Provider.getTokenPrice` has been removed.
+
+- **Changes in methods and parameters:**
+  - `Provider.getBlockWithTransaction` has been replaced with `Provider.getBlock(<BlockTag>, true)`.
+  - `BlockWithTransaction` has been removed.
+
+- **Changes in transaction parameters:**
+  - `TransactionRequest.calldata` has been changed from `BytesLike` to `string`.
+  - `transaction.calldata` parameter in the following methods has been changed from `BytesLike` to `string`:
+    - `Provider.estimateL1ToL2Execute`
+    - `AdapterL1.getRequestExecuteTx`
+    - `AdapterL1.estimateGasRequestExecute`
+    - `AdapterL1.RequestExecute`
+
+- **Utilities:**
+  - `parseTransaction` has been replaced by `Transaction.from`.
```

Comment:
Where does this line come from for the changes? I see `utils.parseTransaction`, this is `parseTransaction`

Reply:
The line you're referring to, utils.parseTransaction, is being replaced by Transaction.from. This change is part of the migration from zksync-web3 to zksync-ethers v6, which involves updating utility functions due to the integration with ethers.js v6.

In zksync-web3, the utility function to parse a transaction was utils.parseTransaction. However, in zksync-ethers v6, this functionality has been updated, and you now use Transaction.from to achieve the same result. This change is necessary to align with the new version's method and parameter changes.

Reply:
That I understand. I'm just wondering why it's `parseTransaction` here instead of `utils.parseTransaction`. Also the old line says it was replaced by `utils.parseEip712`. Is that incorrect?